cluster-api-gcp/deploy/deploy.go

```diff
@@ -33,11 +34,22 @@ type deployer struct {
 	client          *client.ClusterAPIV1Alpha1Client
 }
 
-//it takes path for kubeconfig file.
+// NewDeployer returns a cloud provider specific deployer and
+// sets kubeconfig path for the cluster to be deployed
 func NewDeployer(provider string, configPath string) *deployer {
 	token := util.RandomToken()
 	if configPath == "" {
-		configPath = apiutil.GetDefaultKubeConfigPath()
+		configPath = os.Getenv("KUBECONFIG")
+		if configPath == "" {
+			configPath = apiutil.GetDefaultKubeConfigPath()
+		}
+	} else {
+		// This is needed for kubectl commands run later to create secret in function
```

Comment:
It's unfortunate that we need this, but I'm sure it works. It would be much better to stop shelling out to kubectl and invoke the client libraries to talk to the apiserver directly.